### PR TITLE
refactor: sfx actions

### PR DESCRIPTION
**Pull request by ahkohd**
# Summary of changes

**_Summarize_** the description of your changes.

If it fixes a bug or creates a feature, link the issue.

# Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

## Dependencies

- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any teams which manage the dependencies have been notified of the breaking changes

## Bug (non-breaking change which fixes an issue):

- [ ] Have you **_written tests_** to protect against future occurrences of the bug?

## Feature (non-breaking change which adds functionality)

- [ ] Have you **_seen it working_** in a development environment?
- [ ] Have you **_written tests_** for **_happy_** and **_unhappy_** paths?
- [ ] Have you implemented this feature as per the **_issue acceptance criteria_**?

### Substrate

If this change involves substrate, have you remembered:

- [ ] Storage Migration?
- [ ] RPC methods?
- [ ] Chainspec, both JSON specs & the module?
- [ ] Runtime versioning?
- [ ] Benchmarks?

## Breaking change (a feature that would cause existing functionality not to work as expected)

- [ ] Is the breaking change **_documented_**?
- [ ] Are your commits fully representative of the change?
- [ ] Has any previous functionality been **_deprecated_** and versioned?

## CI/CD

- [ ] If introducing new actions, have you **_looked at the action code_** for anything spurious?
- [ ] Have you written **_custom scripts_** for things that could be actions already on the marketplace?
- [ ] If introducing new actions, have you **_pegged a static version_**?

## Documentation Update

- [ ] Have you checked other documentation, such as the docs repository?
- [ ] If updating script documentation, have you **_tested it on both environments_**?

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

### Summary by OpenAI

Release Notes:

- Refactor: Changes to `mapEncodedArgs` function and `SideEffectSchema` in `sfx.ts` file introduce an enum for the `action` parameter instead of a string. The `SideEffectSchema` now uses an enum validation for the `action` field and adds a new type `SideEffectAction` and a constant object `SideEffectActions` that maps to the possible values of `action`.

> "Enums bring clarity,
> Strings can be quite scary.
> With SideEffectActions,
> Bugs will face subtractions."

### CODE REVIEW

File: client/packages/cli/src/schemas/sfx.ts [hunk 1]
```diff
@@ -80,3 +73,9 @@ export const SideEffectSchema = z.object({
 })
 
 export type SideEffect = z.infer<typeof SideEffectSchema>
+
+export type SideEffectAction = SideEffect["action"]
+export const SideEffectActions = {
```
Comment (MaciejBaj) [inline client/packages/cli/src/schemas/sfx.ts:78]: how about adding full coverage from types/abi/src/standard.rs?

pub fn standard_sfx_abi() -> Vec<(Sfx4bId, SFXAbi)> {
    vec![
        (*b"data", get_data_abi()),
        (*b"tran", get_sfx_transfer_abi()),
        (*b"tass", get_sfx_transfer_asset_abi()),
        (*b"swap", get_swap_abi()),
        (*b"aliq", get_add_liquidity_abi()),
        (*b"rliq", get_remove_liquidity_abi()),
        (*b"cevm", get_call_evm_contract_abi()),
        (*b"wasm", get_call_wasm_contract_abi()),
        (*b"cgen", get_call_generic_abi()),
    ]
}

Comment (ahkohd) [reply client/packages/cli/src/schemas/sfx.ts:78]: I'll make sure to update the schema accordingly!

Comment (ahkohd) [reply client/packages/cli/src/schemas/sfx.ts:88]: Done; see commit [5375fe1](https://github.com/t3rn/t3rn/pull/981/commits/5375fe1d6929c9775cec0af8a8d60e403ad7af64).

**Review by palozano — APPROVED** [review]
LGTM 🚀

**Pull request merged**